### PR TITLE
Programs

### PR DESCRIPTION
**Pull request by ali-gamma**
All 7 programs from trello

### CODE REVIEW

File: program_1.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#1. Write a Ruby program to check whether a given value appears everywhere in a given array. A value is "everywhere" in an array if it presents for #every pair of adjacent elements in the array.
+
+array_1 = []
+while true
+	puts "enter array elements"
+	ele = gets.chop.to_s
+	if ele ==""
+		break
+	end
+	array_1.push(ele)
```
Comment (kalpana19sharma) [inline program_1.rb]: Fix indentation.

File: program_1.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+end
+
+puts "Enter element to search"
+x = gets.chop.to_s
+i=0
+for i in 0...array_1.length-1
```
Comment (kalpana19sharma) [inline program_1.rb]: Use spaces around operators.

File: program_1.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+end
+
+puts "Enter element to search"
+x = gets.chop.to_s
+i=0
+for i in 0...array_1.length-1 
+	if array_1[i] != x && array_1[i+1] != x
+		puts "The element is not everywhere"
+		break
+
```
Comment (kalpana19sharma) [inline program_1.rb]: Fix indentation.

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]
@ali-gamma please fix the indentation and follow the style-guide. Also please use significant file names that show their working/responsibility

File: program_1.rb (hunk 2 above)
Comment (ali-gamma) [reply program_1.rb]: Sure thing ma'am

File: program_1.rb (hunk 1 above)
Comment (ali-gamma) [reply program_1.rb]: How should i indent them? I usually use the tab key once.

File: array_subtract.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#Write a Ruby program to find the difference between the largest and smallest values of a given array of integers and length 1 or more.
+
+array_1 = []
+puts "Enter array"
+while true
+  ele = gets.chomp.to_i
+  if ele == 0
+    break
+  end
```
Comment (kalpana19sharma) [inline array_subtract.rb]: Use single line syntax

File: array_subtract.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  array_1.push(ele)
+end
+if array_1.length == 1
+  puts array_1[0]
+else
+  array_1.sort
+  diff = array_1[-1] - array_1[0]
+  puts "The difference is  :  " + diff.to_s
+
```
Comment (kalpana19sharma) [inline array_subtract.rb]: Remove this unnecessary empty line.

File: character_counter.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+
+array_1 = []
+hash_1 = {}
+while true
+  puts "enter an element"
+  ele = gets.chop.to_s
+  if ele == ""
+    break
+  end
```
Comment (kalpana19sharma) [inline character_counter.rb]: Use single line syntax

File: character_counter.rb [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  array_1.push(ele)
+  if hash_1[ele]
+    hash_1[ele] +=1
+  else
+    hash_1[ele] = 1	
+  end
```
Comment (kalpana19sharma) [inline character_counter.rb]: You can refactor this code using ternary operator.

File: check_everywhere.rb [hunk 8]
```diff
@@ -0,0 +1,70 @@
+#1. Write a Ruby program to check whether a given value appears everywhere in a given array. A value is "everywhere" in an array if it presents for #every pair of adjacent elements in the array.
+
+array_1 = []
+while true
+  puts "enter array elements"
+  ele = gets.chop.to_s
+  if ele ==""
+    break
+  end
```
Comment (kalpana19sharma) [inline check_everywhere.rb]: Use single line syntax

File: double_array.rb [hunk 9]
```diff
@@ -0,0 +1,36 @@
+#Use the map function to double all the elements in the array. However, handle edge cases (like array can have a character) as well.
+
+array_1 = []
+while true
+  ele = gets.chomp
+  if ele == ""
+    break
+  end
```
Comment (kalpana19sharma) [inline double_array.rb]: Use single line syntax

File: double_array.rb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+  array_1.push(ele)
+end
+array_2 = array_1.map{ |n|
+  if n.match( '\d+' )
+    n = 2 * n.to_i
+  else
+    n = n + n
+  end }
```
Comment (kalpana19sharma) [inline double_array.rb]: For multi-line code use `do..end` syntax

File: file_extension.rb [hunk 11]
```diff
@@ -0,0 +1,26 @@
+#3. Write a Ruby program to accept a filename from the user and print the extension of that.
+puts "Enter File Name   "
+name = gets.chop.to_s
+
+i = name.index(".")
+if i == 0 
+  puts "No Extension"
+end
```
Comment (kalpana19sharma) [inline file_extension.rb]: Use single line syntax

File: file_handling.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+#Read from a CSV file, multiple columns (minimum 2), and then write back to the CSV file.
+
+cs = File.open("addresses.csv",'r+')
+lines = cs.readlines
+for line in lines
+  puts line
+end
+new = [ "Ali" , "Noor" , "NA" , "NA" , "NA" ]
```
Comment (kalpana19sharma) [inline file_handling.rb]: Use `%w` syntax for array of strings
https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#collections

File: file_handling.rb [hunk 13]
```diff
@@ -0,0 +1,21 @@
+#Read from a CSV file, multiple columns (minimum 2), and then write back to the CSV file.
+
+cs = File.open("addresses.csv",'r+')
```
Comment (kalpana19sharma) [inline file_handling.rb]: Add space after `,`

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: character_counter.rb [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+
+array_1 = []
+hash_1 = {}
+while true
+  puts "enter an element"
+  ele = gets.chop.to_s
+  break if ele ==""
+  array_1.push(ele)
+  hash_1[ele] = hash_1[ele]? hash_1[ele] + 1 : 1
```
Comment (kalpana19sharma) [inline character_counter.rb]: Is it working?

File: character_counter.rb [hunk 15]
```diff
@@ -0,0 +1,49 @@
+# 2. Using hash table, print the frequency of occurrence of each character inside an array . 
+
+
+
+array_1 = []
+hash_1 = {}
+while true
+  puts "enter an element"
+  ele = gets.chop.to_s
+  break if ele ==""
```
Comment (kalpana19sharma) [inline character_counter.rb]: Add space after `==`

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Comment by ali-gamma** [comment]
Even i was doubtful when i did the correction,but it was working and gave
the same output

File: check_everywhere.rb [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+for i in 0...array_1.length-1 
+  if array_1[i] != x && array_1[i+1] != x
+    puts "The element is not everywhere"
+    break		
+  end
+end
+if i == array_1.length-2
+  puts "Element is everywhere!"
+end	
+=begin
```
Comment (kalpana19sharma) [inline check_everywhere.rb]: Add spaces around operators.

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: check_everywhere.rb [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+puts "Enter element to search"
+x = gets.chop.to_s
+i = 0
+for i in 0 ... array_1.length - 1 
+  if array_1[i] != x && array_1[i+1] != x
+    puts "The element is not everywhere"
+    break		
+  end
+end
+if i == array_1.length-2
```
Comment (kalpana19sharma) [inline check_everywhere.rb]: ???

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**